internal/olm/operator/registry/fbcindex/fbc_registry_pod.go

```diff
@@ -254,13 +261,44 @@ func (f *FBCRegistryPod) podForBundleRegistry(cs *v1alpha1.CatalogSource) (*core
 			},
 		})
 
-		volumeMounts = append(volumeMounts, corev1.VolumeMount{
-			Name:      k8sutil.TrimDNS1123Label(cm.Name + "-volume"),
+		if f.cfg.GzipCM {
+			volumes = append(volumes, corev1.Volume{
+				Name: k8sutil.TrimDNS1123Label(cm.Name + "-unzip"),
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{},
+				},
+			})
+		}
+
+		postfix := cm.Name + "-volume"
+		if f.cfg.GzipCM {
+			postfix = "-unzip"
+		}
+
+		vm := corev1.VolumeMount{
+			Name:      k8sutil.TrimDNS1123Label(cm.Name + postfix),
 			MountPath: path.Join(f.FBCIndexRootDir, cm.Name),
 			SubPath:   cm.Name,
-		})
+		}
+
+		if !f.cfg.GzipCM {
+			vm.SubPath = cm.Name
+		}
```

Comment:
The `VolumeMount.SubPath` is already `cm.Name`. I would remove this if block all together